core/src/main/scala/com/salesforce/op/stages/impl/feature/SmartTextVectorizer.scala

```diff
@@ -77,6 +93,120 @@ class SmartTextVectorizer[T <: Text](uid: String = UID[SmartTextVectorizer[T]])(
     hashSpaceStrategy = getHashSpaceStrategy
   )
 
+  /* CODE FOR DETECTING SENSITIVE FEATURES BEGIN */
+  var guardCheckResults: Option[Array[Boolean]] = None
+
+  override def fit(dataset: Dataset[_]): SequenceModel[T, OPVector] = {
+    /* Set instance variable for guardCheck results here.
+    We compute guardChecks here because all of the logical checks can be implemented efficiently in native Spark,
+    and we would like to use the individual Text columns before they are combined in SequenceEstimator.fit().
+    It is not directly possible to do this computation in the treeAggregate() call below in fitFn() because one of the
+    logical checks computes a standard deviation, which requires knowing the mean beforehand. We could implement the
```

Comment:
Standard deviation can be approximated in a single pass - https://github.com/twitter/algebird/blob/develop/algebird-core/src/main/scala/com/twitter/algebird/MomentsGroup.scala